src/bcrhp/bcrhp/migrations/947_historic_site_one_line.py

```diff
@@ -0,0 +1,368 @@
+                select resourceinstanceid,
+                       __arches_get_concept_label(unnest(heritage_theme)) heritage_theme
+                from heritage_site.heritage_theme group by resourceinstanceid,
+                                                              __arches_get_concept_label(unnest(heritage_theme));
+            create index mv_ht_idx on mv_heritage_theme(resourceinstanceid);
+            commit;
+            """,
+            """
+            drop index if exists mv_ht_idx;
+            drop materialized view if exists mv_heritage_theme cascade;
+            """),
+        RunPrivilegedSQL(
+            """
+            begin;
+            drop index if exists mv_ca_idx;
+            drop materialized view if exists mv_construction_actors cascade;
+            create materialized view mv_construction_actors as
+                select resourceinstanceid,
+                    __arches_get_concept_label(construction_actor_type) actor_type,
+                    array_agg(construction_actor->'en'->>'value') actors,
+                    jsonb_array_elements_text(jsonb_agg(construction_actor_notes->'en'->>'value') - '') notes
+                from heritage_site.construction_actors
+                group by resourceinstanceid, __arches_get_concept_label(construction_actor_type);
+            create index mv_ca_idx on mv_construction_actors(resourceinstanceid);
+            commit;
+            """,
+            """
+            drop index if exists mv_ca_idx;
+            drop materialized view if exists mv_construction_actors cascade;
+            """),
+        RunPrivilegedSQL(
+            """
+            create or replace procedure refresh_materialized_views() as
+            $$
+            BEGIN
+                refresh materialized view mv_bc_right;
+                refresh materialized view mv_bc_statement_of_significance;
+                refresh materialized view mv_borden_number;
+                refresh materialized view mv_chronology;
+                refresh materialized view mv_construction_actors;
+                refresh materialized view mv_government;
+                refresh materialized view mv_heritage_function;
+                refresh materialized view mv_heritage_class;
+                refresh materialized view mv_heritage_theme;
+                refresh materialized view mv_legal_description;
+                refresh materialized view mv_property_address;
+                refresh materialized view mv_protection_event;
+                refresh materialized view mv_geojson_geoms;
+                refresh materialized view mv_site_boundary;
+                refresh materialized view mv_site_names;
+                refresh materialized view mv_site_protection_event;
+                refresh materialized view mv_site_record_admin;
+            END
+            $$ language plpgsql;
+            """,
+            """
+            drop procedure if exists refresh_materialized_views;
```

Comment:
This procedure existed before to refresh the DataBC view. This should reinstate the previous version rather than dropping it.